src/net.cpp

```diff
@@ -649,16 +644,19 @@ void CNode::copyStats(CNodeStats &stats)
 }
 #undef X
 
-// requires LOCK(cs_vRecvMsg)
 bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete)
 {
+    LOCK(cs_vRecvMsg);
     complete = false;
+    int64_t nTime = GetTimeMicros();
+    nLastRecv = nTime / 1000;
```

Comment:
I completely missed that this was a full-second value. Thanks for catching this (twice!). Will fix.